aws_topology/stackstate_checks/aws_topology/resources/stepfunction.py

```diff
@@ -0,0 +1,344 @@
+            self.process_activity(activity)
+
+    @set_required_access_v2('states:ListStateMachines', ignore=True)
+    def process_state_machines(self):
+        for state_machine in self.collect_state_machines():
+            self.process_state_machine(state_machine)
+
+    def process_activity(self, data):
+        output = make_valid_data(data.activity)
+        activity = Activity(data.activity, strict=False)
+        activity.validate()
+        output["tags"] = data.tags
+        self.emit_component(activity.activityArn, 'aws.stepfunction.activity', output)
+
+    def process_state_machine(self, data):
+        # generate component
+        state_machine = StepFunction(data.state_machine, strict=False)
+        output = make_valid_data(data.state_machine)
+        if 'definition' in output:
+            output.pop('definition')
+        output["tags"] = data.tags
+        self.emit_component(state_machine.stateMachineArn, self.COMPONENT_TYPE, output)
+        if state_machine.roleArn:
+            self.agent.relation(state_machine.stateMachineArn, state_machine.roleArn, 'uses service', {})
+        self.process_state_machine_relations(state_machine.stateMachineArn, state_machine.definition)
+
+    def process_state_machine_relations(self, arn, definition):
+        data = json.loads(definition)
+        states = data.get('States')
+        start_state = data.get('StartAt')
+        if states:
+            for state_name, state in states.items():
+                self.process_state(arn, arn, state_name, state, start_state == state_name)
+
+    def process_state(self, sfn_arn, arn, state_name, state, is_start=False):
+        state_arn = sfn_arn + ':state/' + state_name
+        if is_start:
+            self.agent.relation(arn, state_arn, 'uses service', {})  # starts state
+        state["Name"] = state_name
+        next_state = state.get('Next')
+        if next_state:
+            next_state_arn = sfn_arn + ':state/' + next_state
+            self.agent.relation(state_arn, next_state_arn, 'uses service', {})  # can transition to
+        state_type = state.get('Type')
+        if state_type == 'Choice':
+            self.process_choice_state(sfn_arn, state_arn, state.get('Choices'), state.get('Default'))
+        elif state_type == 'Map':
+            self.process_map_state(sfn_arn, state_arn, state.get('Iterator'))
+        elif state_type == 'Parallel':
+            self.process_parallel_state(sfn_arn, state_arn, state.get('Branches'))
+        elif state_type == 'Task':
+            self.process_task_state(state_arn, state)
+        self.emit_component(state_arn, 'aws.stepfunction.state', state)
+
+    def process_choice_state(self, sfn_arn, state_arn, choices, default_state):
+        choices = choices or []
+        if default_state:
+            default_state_arn = sfn_arn + ':state/' + default_state
+            self.agent.relation(state_arn, default_state_arn, 'uses service', {})  # can transition to (default)
```

Comment:
Can we rename `self.agent.relation` to ` self.emit_relation` to keep the interface stable?

Reply:
This will be on separate ticket/branch